codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -264,27 +269,35 @@ public ChannelFuture writeSettingsAck(ChannelHandlerContext ctx, ChannelPromise
     }
 
     @Override
-    public ChannelFuture writePing(ChannelHandlerContext ctx, boolean ack, ByteBuf data,
-            ChannelPromise promise) {
-        boolean releaseData = true;
-        SimpleChannelPromiseAggregator promiseAggregator =
+    public ChannelFuture writePing(ChannelHandlerContext ctx, boolean ack, ByteBuf data, ChannelPromise promise) {
+        final SimpleChannelPromiseAggregator promiseAggregator =
```

Comment:
it is assumed this operation won't throw ... it makes the code simpler and if object allocation fails (of an internal object) we are very unlikely to be able to make progress. However if folks prefer the code can account for this operation throwing too.